readme.html

```diff
@@ -114,6 +117,117 @@ <h2>Using the Plugin</h2>
+    will then upload the data using a HTTP PUT request. After the upload has succeeded, the client will send, over XMPP,
+    send the URL to the intended recipients of the data. Their clients will receive that URL (over XMPP) and subsequently
+    perform a HTTP GET request to download the data from the web server.
+</p>
+
+<h3>Configuration A: Using multiple service endpoint addresses</h3>
+
+<p>
+    When each of the Openfire cluster node servers is connected directly to the internet, accessible to clients through
+    their fully qualified domain name, then no configuration might be needed to make HTTP File Upload functionality
+    available on all cluster nodes, for all clients.
+</p>
+
+<h4>Option 1: Each Openfire manages a unique set of data</h4>
+
+<p>
+    <em>pro's & cons:</em> This solution requires no configuration, provides adequate scalability, but does not offer
+    much data redundancy or flexibility in network architecture.
+</p>
+
+<p>
+    Since version 1.2.0 of this plugin, the definition of a Slot is shared across all XMPP cluster nodes. This slot
+    defines the endpoint addresses to be used by the uploader, as well as the recipients of the data.
+</p>
+
+<p>
+    In the scenario described by this configuration option, each Openfire cluster node will issue slots that contain
+    endpoint URLs that are specific to the server that issued the slot. This means that the upload, as well as all
+    downloads, will be directed to URL on that same server.
+</p>
+
+<h3>Configuration B: Using one service endpoint address</h3>
+
+<p>
+    In certain environments it is undesirable or unpractical to expose more than one endpoint URL to chat clients. This
+    is frequently the case when the network architecture includes a load balancer. For this or other reasons, it might
+    have been necessary to reconfigure the 'announced' endpoints (using the configuration options defined in the
+    <a href="#configuration">Configuration paragraph</a> of this document) to use the same endpoint address for all
+    requests.
+</p>
+
+<h4>Option 2: Use only one server for file upload/download</h4>
+
+<p>
+    <em>pro's & cons:</em> This solution is typically easy to configure, but provides poor redundancy and
+    scalability.
+</p>
+
+<p>
+    In this scenario, the singular endpoint address could be mapped to exactly one Openfire cluster node. This can be
+    achieved by directly mapping the 'announced' endpoint details to the address of one of the Openfire nodes,
+    (as described in the <a href="#configuration">Configuration paragraph</a> of this document) or by configuring the
+    load balancer to 'balance' the traffic to just one of the nodes.
+</p>
+
+<h4>Option 3: Use every cluster node for file upload/download</h4>
+
+<p>
+    <em>pro's & cons:</em> More complex to configure, but provides better redundancy and scalability.
```

Comment:
Apostrophes again. I should've just batched this all together as a commit :facepalm: 

```suggestion
    <em>pros & cons:</em> More complex to configure, but provides better redundancy and scalability.
```